Parameter "chromatic" set to True causes failure of SSIM if greyscale image

**Describe the bug**
In FSIM, if option "chromatic" is set to True the algo will try to compute FSIMc regardless of whether the images has the sufficient number of channels to do so.

**To Reproduce**
If I write:
```
    # Greyscale images
    goldhill = torch.tensor(imread('tests/assets/goldhill.gif'))
    goldhill_jpeg = torch.tensor(imread('tests/assets/goldhill_jpeg.gif'))

    score = fsim(goldhill_jpeg, goldhill, data_range=255, chromatic=True, reduction='none')
```
I get the error:
```
        # Compute FSIM
        PC = similarity_map(pc_x, pc_y, T1)
        GM = similarity_map(grad_map_x, grad_map_y, T2)
        pc_max = torch.where(pc_x > pc_y, pc_x, pc_y)
        score = GM * PC * pc_max
    
        if chromatic:
>           S_I = similarity_map(x_i, y_i, T3)
E           UnboundLocalError: local variable 'x_i' referenced before assignment

piq/fsim.py:110: UnboundLocalError

```

**Expected behavior**
Like in GMSD, put an asserting that the chromatic option can only be used the case where there are 3 ndims.

--- a/piq/fsim.py
+++ b/piq/fsim.py
@@ -107,6 +107,7 @@ def fsim(x: torch.Tensor, y: torch.Tensor, reduction: str = 'mean',
     score = GM * PC * pc_max
     
     if chromatic:
+        assert num_channels == 3, "Chromatic component can be computed only for RGB images!"
         S_I = similarity_map(x_i, y_i, T3)
         S_Q = similarity_map(x_q, y_q, T4)
         score = score * torch.abs(S_I * S_Q) ** lmbda

--- a/tests/test_fsim.py
+++ b/tests/test_fsim.py
@@ -24,20 +24,35 @@ def target() -> torch.Tensor:
 
 @pytest.fixture(scope='module')
 def prediction_grey() -> torch.Tensor:
-    return torch.rand(3, 3, 256, 256)
+    return torch.rand(3, 1, 256, 256)
 
 
 @pytest.fixture(scope='module')
 def target_grey() -> torch.Tensor:
-    return torch.rand(3, 3, 256, 256)
+    return torch.rand(3, 1, 256, 256)
 
 
 # ================== Test function: `fsim` ==================
+def test_fsim_forward(input_tensors, device: str) -> None:
+    prediction, target = input_tensors
+    fsim(prediction.to(device), target.to(device))
+
+
 @pytest.mark.parametrize("chromatic", [False, True])
 def test_fsim_symmetry(prediction: torch.Tensor, target: torch.Tensor, chromatic: bool, device: str) -> None:
-    measure = fsim(prediction, target, data_range=1., chromatic=chromatic)
-    reverse_measure = fsim(target, prediction, data_range=1., chromatic=chromatic)
+    measure = fsim(prediction.to(device), target.to(device), data_range=1., chromatic=chromatic)
+    reverse_measure = fsim(target.to(device), prediction.to(device), data_range=1., chromatic=chromatic)
     assert (measure == reverse_measure).all(), f'Expect: FSIM(a, b) == FSIM(b, a), got {measure} != {reverse_measure}'
+
+
+@pytest.mark.parametrize(
+    "chromatic,expectation",
+    [(False, raise_nothing()),
+     (True, raise_nothing())])
+def test_fsim_chromatic_raises_for_greyscale(
+        prediction_grey: torch.Tensor, target_grey: torch.Tensor, chromatic: bool, expectation: Any) -> None:
+    with expectation:
+        fsim(prediction_grey, target_grey, data_range=1., chromatic=chromatic)
 
 
 @pytest.mark.parametrize(
@@ -111,6 +126,6 @@ def test_fsim_loss_reduction(prediction: torch.Tensor, target: torch.Tensor) -> 
 
 def test_fsim_loss_computes_grad(prediction: torch.Tensor, target: torch.Tensor, device: str) -> None:
     prediction.requires_grad_()
-    loss_value = FSIMLoss()(prediction, target)
+    loss_value = FSIMLoss()(prediction.to(device), target.to(device))
     loss_value.backward()
     assert prediction.grad is not None, 'Expected non None gradient of leaf variable'

--- a/tests/test_fsim.py
+++ b/tests/test_fsim.py
@@ -35,7 +35,7 @@ def target_grey() -> torch.Tensor:
 # ================== Test function: `fsim` ==================
 def test_fsim_forward(input_tensors, device: str) -> None:
     prediction, target = input_tensors
-    fsim(prediction.to(device), target.to(device))
+    fsim(prediction.to(device), target.to(device), chromatic=False)
 
 
 @pytest.mark.parametrize("chromatic", [False, True])
@@ -48,7 +48,7 @@ def test_fsim_symmetry(prediction: torch.Tensor, target: torch.Tensor, chromatic
 @pytest.mark.parametrize(
     "chromatic,expectation",
     [(False, raise_nothing()),
-     (True, raise_nothing())])
+     (True, pytest.raises(AssertionError))])
 def test_fsim_chromatic_raises_for_greyscale(
         prediction_grey: torch.Tensor, target_grey: torch.Tensor, chromatic: bool, expectation: Any) -> None:
     with expectation:

